frappe_graphql/utils/resolver/list_resolver.py

```diff
@@ -0,0 +1,131 @@
+import base64
+from graphql import GraphQLResolveInfo, GraphQLError
+
+import frappe
+from frappe.model.db_query import DatabaseQuery
+
+
+def list_resolver(obj, info: GraphQLResolveInfo, **kwargs):
+    doctype = kwargs["doctype"]
+    frappe.has_permission(doctype=doctype, throw=True)
+
+    before = kwargs.get("before")
+    after = kwargs.get("after")
+    first = kwargs.get("first")
+    last = kwargs.get("last")
+    filters = kwargs.get("filter") or []
+    sort_key, sort_dir = get_sort_args(kwargs.get("sortBy"))
+    _validate_connection_args(kwargs)
+
+    original_sort_dir = sort_dir
+    if last and sort_dir == "asc":
+        # to get LAST, we swap the sort order
+        # data will be reversed after fetch
+        sort_dir = "desc"
+
+    cursor = after or before
+    limit = first or last
+
+    filters = process_filters(doctype, filters)
+
+    count = frappe.db.sql(f"""
```

Comment:
why not use get_list() ? best to use this because permissions (is_owner in particular) are handled in get_list or else you will have to handle it in your sql

Reply:
oops, i missed the permissions.